server/src/main/java/org/opensearch/cluster/coordination/PublishWithManifestRequest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.coordination;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.gateway.remote.ClusterMetadataManifest;
+
+import java.util.Objects;
+
+/**
+ * This class represents a publish request that includes a {@link ClusterMetadataManifest}.
+ * It extends the {@link PublishRequest} class and adds a field for the accepted manifest.
+ */
+public class PublishWithManifestRequest extends PublishRequest {
```

Comment:
I dont think, we need to add a new class

Reply:
I wanted to avoid the null check of manifest, and opted for type check. Will add the manifest to PublishRequest only.

Reply:
There is RemotePublishRequest class, I dont want to change PublishRequest class

Reply:
Removed this class